src/atlannot/evaluation.py

```diff
@@ -0,0 +1,221 @@
+
+    Returns
+    -------
+    results: dict[int, float]
+        Dictionary containing the region id as keys and the mean of the
+        jaggedness of that given region id as values.
+    """
+    metrics = core.compute(volume, coronal_axis_index=axis, regions=region_ids)
+    if region_ids is None:
+        return {
+            region_id: values["mean"]
+            for region_id, values in metrics["perRegion"].items()
+        }
+
+    results = {}
+    for region_id in region_ids:
+        results[region_id] = metrics["perRegion"][region_id]["mean"]
+    return results
+
+
+def compute_iou(vol_true, vol_pred, region_ids=None):
+    """Compute the intersection over union of given region IDs.
+
+    Parameters
+    ----------
+    vol_true: np.ndarray
+        Input volume considered as ground truth.
+    vol_pred: np.ndarray
+        Input volume considered as prediction.
+    region_ids: list[int] | None
+        List of region IDs to compute the intersection over union.
+        If None, the IoU is computed for all the region IDs present in the volume.
+
+    Returns
+    -------
+    results: dict[int, float]
+        Dictionary containing the region id as keys and the average of the
+        intersection over union of that given region id as values.
+    """
+    results = {}
+    if region_ids is None:
+        labels = np.unique(vol_true)
+        for label in labels:
+            results[label] = iou_score(vol_true, vol_pred, k=label)[0]
+        return results
+
+    for region_id in region_ids:
+        results[region_id] = iou_score(vol_true, vol_pred, k=region_id)[0]
+
+    return results
+
+
+def compute_region_entropy(nissl, atlas, label_value, bins=100):
+    """Compute the entropy of densities of Nissl for a given label value in the atlas.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+    label_value: int
+        Label Value to consider.
+    bins: int
+        Number of bins used to compute the histogram of the densities.
+
+    Returns
+    -------
+    region_entropy: float
+        Entropy of the densities of Nissl at a given region value.
+    """
+    prob_region, _ = np.histogram(nissl[atlas == label_value], bins=bins)
+    region_entropy = entropy(prob_region)
+    return region_entropy
+
+
+def compute_entropies(nissl, atlas):
+    """Compute entropies of Nissl densities.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+
+    Returns
+    -------
+    brain_entropy: float
+        Entropy of the densities of Nissl for the entire brain.
+
+    conditional_entropy: float
+        Conditional entropy of the densities of Nissl depending on the brain regions.
+    """
+    # 1: For the entire brain
+    prob_entire_brain, _ = np.histogram(nissl[atlas != 0], bins=100)
+    brain_entropy = entropy(prob_entire_brain)
+
```

Comment:
Maybe this function can focus on computing the conditional entropies and not the total entropy of the brain.

I see that the region mask is slightly different, and that's why `compute_region_entropy` can't be reused. What if we change `compute_region_entropy` slighly like this:
```python
def dist_entropy(data, bins=100):
    """Compute the entropy of the value distribution of data."""
    hist, _ = np.histogram(data, bins=bins)
    return stats.entropy(hist)
```
then one can write `brain_entropy = dist_entropy(nissl[atlas != 0])` and in `compute_entropies` one can write
```python
all_region_entropy.append(dist_entropy(nissl[atlas == label]) * count)
```

Reply:
What do you think about this change baa8bcd ?

Reply:
looks great!!!